go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -135,6 +137,54 @@ func (c *Controller) syncUserDefinedNetwork(udn *userdefinednetworkv1.UserDefine
 		return nil, nil
 	}
 
+	if !udn.DeletionTimestamp.IsZero() { // udn is being  deleted
+		if controllerutil.ContainsFinalizer(udn, template.FinalizerUserDefinedNetwork) {
+
+			if nad != nil &&
+				metav1.IsControlledBy(nad, udn) &&
+				controllerutil.ContainsFinalizer(nad, template.FinalizerUserDefinedNetwork) {
+
+				// TODO: verify no pod is using the NAD before removing finalizer from NAD
+
+				controllerutil.RemoveFinalizer(nad, template.FinalizerUserDefinedNetwork)
+				patch, err := newFinalizersReplaceJsonPatch(nad.Finalizers)
+				if err != nil {
+					return nil, err
+				}
+				nad, err = c.nadClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(nad.Namespace).Patch(context.Background(), nad.Name, types.JSONPatchType, patch, metav1.PatchOptions{})
+				if err != nil {
+					return nil, err
+				}
```

Comment:
one more thing: i think it is safer to just do `Update` instead of Patch here and in the following calls. It will also make sure you only remove your own finalizers, which is  a good tone for controllers

Reply:
Done